sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.students.controller.ts

```diff
@@ -0,0 +1,140 @@
+   * @param studentUserToken student user token to authorize the user.
+   * @returns Application Offering Change Request details.
+   */
+  @Get()
+  @ApiNotFoundResponse({
+    description: "Not able to find an Application Offering Change Request.",
+  })
+  async getById(
+    @Param("applicationOfferingChangeRequestId", ParseIntPipe)
+    applicationOfferingChangeRequestId: number,
+    @UserToken()
+    studentUserToken: StudentUserToken,
+  ): Promise<ApplicationOfferingChangesAPIOutDTO> {
+    return this.applicationOfferingChangeRequestControllerService.getById(
+      applicationOfferingChangeRequestId,
+      { studentId: studentUserToken.studentId },
+    );
+  }
+
+  /**
+   * Gets the Application Offering Change Request status.
+   * @param applicationOfferingChangeRequestId the Application Offering Change Request id.
+   * @param studentUserToken student user token to authorize the user.
+   * @returns Application Offering Change Request status.
+   */
+  @Get("application-offering-change-request-status")
+  @ApiNotFoundResponse({
+    description:
+      "Not able to get the Application Offering Change Request Status.",
+  })
+  async getApplicationOfferingChangeRequestStatusById(
+    @Param("applicationOfferingChangeRequestId", ParseIntPipe)
+    applicationOfferingChangeRequestId: number,
+    @UserToken()
+    studentUserToken: StudentUserToken,
+  ): Promise<ApplicationOfferingChangeRequestStatus> {
+    const applicationOfferingChangeRequestStatus =
+      await this.applicationOfferingChangeRequestService.getApplicationOfferingChangeRequestStatusById(
+        applicationOfferingChangeRequestId,
+        {
+          studentId: studentUserToken.studentId,
+        },
+      );
+    if (!applicationOfferingChangeRequestStatus) {
+      throw new NotFoundException(
+        "Not able to get the Application Offering Change Request Status.",
+      );
+    }
+    return applicationOfferingChangeRequestStatus;
+  }
+
+  /**
+   * Updates an application offering change request status.
+   * @param applicationOfferingChangeRequestId application offering change request id.
+   * @param userToken user token to authorize the user.
+   * @param payload information to update the application offering change request status.
+   */
+  @Patch()
+  @ApiUnauthorizedResponse({
+    description:
+      "The student does not have access to the application offering change request.",
+  })
+  async updateApplicationOfferingChangeRequestStatus(
+    @Param("applicationOfferingChangeRequestId", ParseIntPipe)
+    applicationOfferingChangeRequestId: number,
+    @UserToken()
+    userToken: StudentUserToken,
+    @Body()
+    payload: UpdateApplicationOfferingChangeRequestAPIInDTO,
+  ) {
+    const studentAuthorized =
+      await this.applicationOfferingChangeRequestService.validateStudentForOfferingChangeRequest(
+        applicationOfferingChangeRequestId,
+        userToken.studentId,
+      );
+    if (studentAuthorized)
```

Comment:
pls follow the approach that we do in other controllers are avoid `if else` condition, 
![image](https://github.com/bcgov/SIMS/assets/77353155/fa375881-64bf-44a8-9040-ca66e0035c05)

Reply:
Is that to follow Fail Fast approach? https://en.wikipedia.org/wiki/Fail-fast